c3bottles/views/api.py

```diff
@@ -32,6 +32,30 @@ def process():
     )
 
 
+@bp.route("/api/all_dp.json", methods=("POST", "GET"))
+def all_dp():
+    return dp_json()
+
+
+@bp.route("/api/map_source.json")
+def map_source():
+    map_source = app.config.get('MAP_SOURCE', {})
+    return jsonify({
+        "attribution": map_source.get('attribution', ''),
+        "tileserver": map_source.get('tileserver', ''),
+        "tileserver_subdomains": map_source.get("tileserver_subdomains", []),
+        "bounds": map_source.get("bounds", None),
+        "initial_view": map_source.get("initial_view", None),
+        "level_config": map_source.get("level_config", None),
+        "min_zoom": map_source.get("min_zoom", 0),
+        "max_zoom": map_source.get("max_zoom", 0),
+        "simple_crs": map_source.get("simple_crs", False),
+        "hack_257px": map_source.get("hack_257px", False),
+        "tms": map_source.get("tms", False),
+        "no_wrap": map_source.get("no_wrap", False)
+    })
+
```

Comment:
I like this idea, however I would put the JSON serialization into the `MapSource` class by either making that JSON-serializeable or adding a method there that returns a JSON-serializable object like a dict of primitive values.

The base class would also be a good place to set these default values. I was thinking about it when introducing the map source configuration but didn't do it since the JS part is the only place where we currently use the map sources anyway, so I figured that putting it into the JS would be easier.

Reply:
I do not know best practice here.